### PR TITLE
Added view single task page

### LINKED ISSUE #95: Create view task details template
Create a template that displays full details of a given task

### PR DESCRIPTION
**Pull request by AvivLiberman**
Fixes #95

This PR include view page for single task, for now it is without design.
To make this PR small I have added 2 issues that following it #112, #113
## Changes you made
- Added view_single_task page
- Redirect home if not authenticated
- Path to the page: tasks/<task_id>

### CODE REVIEW

**Review by orzionpour — DISMISSED** [review]
LGTM  👑

**Review by Asafh7 — APPROVED** [review]

**Comment by kobihk** [comment]
@AvivLiberman could you please add tests to see that you get the correct response from the client for each page

**Comment by AvivLiberman** [comment]
> @AvivLiberman could you please add tests to see that you get the correct response from the client for each page
wdym? in the lectures you said we are not testing the UI
@kobihk

**Comment by AvivLiberman** [comment]
@kobihk Done

**Review by avraham1992 — DISMISSED** [review]

**Review by orzionpour — APPROVED** [review]

File: tasks/tests/test_view_single_task.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class TestViewSingleTask:
+
+    def test_no_authenticated_user_asks_task(self, client, task_1):
+        response = client.get(f'/tasks/{task_1.id}')
+        assert response.status_code == 302
+        assert response.url == '/'
+
+    def test_with_employee_logged_in(self, client, task_1, employee_1):
+        client.login(username=employee_1.user.username,
+                     password='password')
```
Comment (kobihk) [inline tasks/tests/test_view_single_task.py:16]: I don't think you need to break it into a few lines
unless its many variables with a long line for each variable:
```suggestion
        client.login(username=employee_1.user.username, password='password')
```

File: tasks/tests/test_view_single_task.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        response = client.get(
+            f'/tasks/{task_1.id}', enforce_csrf_checks=True)
```
Comment (kobihk) [inline tasks/tests/test_view_single_task.py:18]: same ^^
```suggestion
        response = client.get(f'/tasks/{task_1.id}', enforce_csrf_checks=True)
```

File: tasks/tests/test_view_single_task.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        assert response.status_code == 200
+        assert response.context['task'] == task_1
+
+    def test_view_task_form(self, client, task_1, employee_1):
+        client.login(username=employee_1.user.username,
+                     password='password')
```
Comment (kobihk) [inline tasks/tests/test_view_single_task.py:24]: same ^^
```suggestion
        client.login(username=employee_1.user.username, password='password')
```

File: tasks/tests/test_view_single_task.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        response = client.get(f'/tasks/{task_1.id}')
+        assert isinstance(response.context.get('form'), TaskForm)
+
+    def test_change_status(self, client, task_1, employee_1):
+        client.login(username=employee_1.user.username,
+                     password='password')
```
Comment (kobihk) [inline tasks/tests/test_view_single_task.py:30]: same ^^
```suggestion
        client.login(username=employee_1.user.username, password='password')
```

**Review by kobihk — CHANGES_REQUESTED** [review]

**Comment by AvivLiberman** [comment]
@kobihk Done

**Review by kobihk — APPROVED** [review]

**Review by orzionpour — DISMISSED** [review]

**Review by aradz1997 — APPROVED** [review]

**Review by Asafh7 — DISMISSED** [review]

**Comment by mgold1234** [comment]
please fix the conflict and then we merge it:)

**Review by avraham1992 — APPROVED** [review]

**Review by avraham1992 — CHANGES_REQUESTED** [review]
It is not possible to see the tasks if you do not log in to the app so here is a problem according to what I see your bar shows that needs to be done login ..
This means that there is a problem here @AvivLiberman  once there is a separation between landing pages and an external page do not do identification checks again.. it happens because you do not perform identifying that there is no need! The identification is done on the login page and this is it .. what I did is enough you sold not do the identification again insid app Arad have too same he changes it next PR

**Comment by AvivLiberman** [comment]
> It is not possible to see the tasks if you do not log in to the app so here is a problem according to what I see your bar shows that needs to be done login .. This means that there is a problem here @AvivLiberman once there is a separation between landing pages and an external page do not do identification checks again.. it happens because you do not perform identifying that there is no need! The identification is done on the login page and this is it .. what I did is enough you sold not do the identification again insid app Arad have too same he changes it next PR

Sorry I don't understand the problem

**Review by orzionpour — APPROVED** [review]

**Review by avraham1992 — APPROVED** [review]

**Review by natifridman — APPROVED** [review]

**Pull request merged**